### ISSUE TITLE
Revisit Checkbox Fields

### ISSUE THREAD

**Issue by dillonredding** [siren-js/client#22]
Currently, `checkbox` fields work similar to HTML.

```json
{
  "type": "checkbox",
  "name": "foo",
  "value": "bar",
  "checked": false
}
```

In the above `Field` example, `foo=bar` will only be sent to the server if `checked` is truthy. Otherwise, `foo` is not sent at all (see [here](https://github.com/siren-js/client/blob/main/src/serialize/fields-to-name-value-pairs.ts#L31)). However, this requires clients to understand and utilize the `checked` extension for submission. I think we should abandon the need for the extension and send the field's `value` regardless. The checkedness can be tracked elsewhere and used to update the field's `value` (e.g., to `true`/`false`).

Basically, `checkbox` fields should have the type `Field<boolean>`.

**Comment by SeeSharpCode** [comment]
Am I understanding correctly that the field would look like this:

```json
{
  "type": "checkbox",
  "name": "foo",
  "value": false
}
```

And `value=false` would be sent to the server?

**Comment by dillonredding** [comment]
The main difference for this library would be removing the `field.checked` condition in the default field serializer. Instead of including `{field.name}={field.value}` or _nothing_ as with HTML, the former would always be included and it would be up to the client what the field value would be when the checkbox is checked vs unchecked. Since it's a checkbox, a `boolean` probably makes the most sense, but it could technically be any values that make sense in the context of the API.

**Comment by SeeSharpCode** [comment]
Sounds good. I'm in favor of this change since it will make checkbox field behavior consistent with other field types.
